factory/error.go

```diff
@@ -0,0 +1,25 @@
+package factory
+
+import (
+	"github.com/juju/errgo"
+)
+
+var (
+	maskAny = errgo.MaskFunc(errgo.Any)
+)
+
+func maskAnyf(cause error, f string, v ...interface{}) error {
+	err := maskAny(errgo.WithCausef(nil, cause, f, v...))
```

Comment:
There were problems with appending the format string in the message. This should be tested.